lib/sidekiq/launcher.rb

```diff
@@ -172,17 +172,15 @@ def ❤
               "quiet", @done.to_s,
               "rss", kb)
             transaction.expire(key, 60)
-            transaction.rpop("#{key}-signals")
+            transaction.rpop("#{key}-signals") unless @embedded
           }
```

Comment:
It's possible to send a signal into Redis for the process and that signal will never get popped, leading to a memory leak. It might be better to pop the signal and change the Process.kill logic below to `if signal && !@embedded`.

Reply:
yes, that makes sense :+1: I'll update.

Reply:
Updated. Please confirm the key that goes in to `processes` is the same as `identity`? I had to update the tests to pass as it was using a different key and identity, but `Launcher` has `key = identity` - so I think this is ok